tools/check_dist.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+
+import sys
+import tomllib
+from zipfile import ZipFile
+
+from pathspec.patterns.gitwildmatch import GitWildMatchPattern
+
+
+def ensure_all_artifacts_included(pyproject, wheel_paths):
+    try:
+        artifacts = pyproject["tool"]["hatch"]["build"]["artifacts"]
+    except KeyError:
+        print("No artifacts specified")
+        return 0
+    status = 0
+    for wheel_path in wheel_paths:
+        with ZipFile(wheel_path) as f:
+            included_files = f.namelist()
+        for artifact in artifacts:
+            pattern = GitWildMatchPattern(artifact)
+            if all(pattern.match_file(name) is None for name in included_files):
+                print(f"{wheel_path}: No file matches artifact: {artifact}")
+                status |= 1
+    return status
+
+
+def main(argv):
+    if len(argv) < 3:
+        print(f"USAGE: {argv[0]} <pyproject.toml path> <.whl path>...")
+        return 1
```

Comment:
I don't like this direct usage of `argv`. I would vote for using https://docs.python.org/3/library/argparse.html.